### PR TITLE
Extract traceEnabled for reuse in RuleBasedTransactionAttribute

### PR DESCRIPTION
**Pull request by chenqimiao**
Fix logger.isTraceEnabled check before the last logger.trace.
And extract an traceEnabled variable for reuse.

### CODE REVIEW

File: spring-tx/src/main/java/org/springframework/transaction/interceptor/RuleBasedTransactionAttribute.java [hunk 1]
```diff
@@ -147,13 +148,15 @@ public boolean rollbackOn(Throwable ex) {
...
-		if (logger.isTraceEnabled()) {
+		if (traceEnabled) {
 			logger.trace("Winning rollback rule is: " + winner);
 		}
 
 		// User superclass behavior (rollback on unchecked) if no rule matches.
 		if (winner == null) {
-			logger.trace("No relevant rollback rule found: applying default rules");
+			if(traceEnabled) {
+				logger.trace("No relevant rollback rule found: applying default rules");
```
Comment (NathanQingyangXu) [inline spring-tx/src/main/java/org/springframework/transaction/interceptor/RuleBasedTransactionAttribute.java]: The above statement won't incur unnecessary perf overhead like in other cases (string concatenation has to be done beforehand) and IMO the logging level checking is not that necessary.

Comment (chenqimiao) [reply spring-tx/src/main/java/org/springframework/transaction/interceptor/RuleBasedTransactionAttribute.java]: Log level check whether it is necessary depends on your logging system.
I think Spring needs this check for compatibility.

Comment (chenqimiao) [reply spring-tx/src/main/java/org/springframework/transaction/interceptor/RuleBasedTransactionAttribute.java]: > The above statement won't incur unnecessary perf overhead like in other cases (string concatenation has to be done beforehand) and IMO the logging level checking is not that necessary.

Log level check whether it is necessary depends on your logging system.
I think Spring needs this check for compatibility.

**Comment by sbrannen** [comment]
Thanks for the PR.

For these particular use cases, we don't think there is much benefit in applying the proposed changes. If trace logging becomes a performance issue, we might revisit whether we need to trace logging at all.

In light of that, I am closing this PR.

**Comment by chenqimiao** [comment]
@sbrannen I get it, thank you for your pointing.

**Pull request closed without merging**